### ISSUE TITLE
Is SentenceTransformer changed when loading from cache?

### ISSUE THREAD

**Issue by PaulForInvent** [UKPLab/sentence-transformers#1038]
Hey,

I am curious, because of the simple assignment and not deep copy for a SentenceTransformer model while fientuning.

I f you use the finetuning usage like:

```
model = SentenceTransformer(model_name)
model.fit(...)
```

The SentenceTransformer(model_name) is also changed because of simple assignment. Now it depends how this is cached.

Up to now I thought using the above syntax I am save to get each time a plain empty fresh laoded model? Using this in a loop (tuning with each time you want a fresh model) or using a new session where the model is loaded from cache, di I get a chnaged model or the fresh default model?

**Comment by nreimers** [comment]
Yes, you get each time the model from disc.

**Comment by PaulForInvent** [comment]
@nreimers this does not answer the question, also, it is an OR question. ;-) Of course it is from disc...
But is it changed or the same as the fresh loaded default model?

**Comment by nreimers** [comment]
Fresh from disc

**Comment by PaulForInvent** [comment]
Yes, of course, but my issue was another one. Is this disc version changed due to the simple assignment?

**Comment by nreimers** [comment]
No

**Comment by PaulForInvent** [comment]
Ok. Since the simple assigment modifies the SentenceTransformer object I suspect this is then cached on disk, and later when you load the SentenceTransformer(path_from_disc) via a path, you load this modified object, and not the original downlaoded model from Huggingface?
